ledger/src/get.rs

```diff
@@ -23,7 +23,30 @@ impl<N: Network, C: ConsensusStorage<N>> Ledger<N, C> {
 
     /// Returns the committee for the given `round`.
     pub fn get_committee_for_round(&self, round: u64) -> Result<Option<Committee<N>>> {
-        self.vm.finalize_store().committee_store().get_committee_for_round(round)
+        // Check if the committee is already in the cache.
+        if let Some(committee) = self.committee_cache.lock().get(&round) {
+            return Ok(Some(committee.clone()));
+        }
+
+        match self.vm.finalize_store().committee_store().get_committee_for_round(round)? {
+            // Return the committee if it exists.
+            Some(committee) => {
+                // Insert the committee into the cache.
+                self.committee_cache.lock().push(round, committee.clone());
+                // Return the committee.
+                Ok(Some(committee))
+            }
+            // Return the current committee if the round is equivalent.
+            None => {
+                // Retrieve the current committee.
+                let current_committee = self.latest_committee()?;
```

Comment:
For future reviewers: this additional logic came from the snarkOS side: https://github.com/ProvableHQ/snarkOS/pull/3526/

I think at least its good that all calls to `get_committee_for_round` now use the same logic...